client/blocks/video-editor/index.jsx

```diff
@@ -0,0 +1,225 @@
+import Notice from 'components/notice';
+import DetailPreviewVideo from 'post-editor/media-modal/detail/detail-preview-video';
+import VideoEditorButtons from './video-editor-buttons';
+import {
+	resetVideoEditorPosterState,
+	resetVideoEditorState,
+	updateVideoEditorPoster,
+} from 'state/ui/editor/video-editor/actions';
+import {
+	getVideoEditorPoster,
+	isVideoEditorPosterUpdated,
+	isVideoEditorPosterUpdating,
+	isVideoEditorVideoLoaded,
+	videoEditorHasPosterUpdateError,
+	videoEditorHasScriptLoadError,
+} from 'state/ui/editor/video-editor/selectors';
+
+/**
+ * Module variables
+ */
+let isSelectingFrame = false;
+
+class VideoEditor extends Component {
+	static propTypes = {
+		className: PropTypes.string,
+		media: PropTypes.object.isRequired,
+		onCancel: PropTypes.func,
+		onUpdatePoster: PropTypes.func,
+
+		// Connected props
+		hasPosterUpdateError: PropTypes.bool,
+		hasScriptLoadError: PropTypes.bool,
+		isPosterUpdated: PropTypes.bool,
+		isPosterUpdating: PropTypes.bool,
+		isVideoLoaded: PropTypes.bool,
+		poster: PropTypes.string,
+	};
+
+	static defaultProps = {
+		onCancel: noop,
+		onUpdatePoster: noop,
+	};
+
+	state = {
+		pauseVideo: false,
+		videoError: false,
+	};
+
+	componentDidMount() {
+		this.props.resetState();
+	}
+
+	shouldComponentUpdate( nextProps ) {
+		if ( nextProps.isPosterUpdated && ! nextProps.hasPosterUpdateError ) {
+			this.props.onUpdatePoster( this.getVideoEditorProps( nextProps.poster ) );
+
+			return false;
+		}
+
+		return true;
+	}
+
+	handleSelectFrame = () => {
+		const { isVideoLoaded } = this.props;
+
+		this.props.resetPosterState();
+
+		if ( ! isVideoLoaded ) {
+			this.setState( { videoError: true } );
+			return;
+		}
+
+		isSelectingFrame = true;
+		this.setState( { pauseVideo: true } );
+	}
+
+	handlePause = ( currentTime ) => {
+		if ( ! isSelectingFrame ) {
+			return;
+		}
+
+		const {
+			media,
+			updatePoster,
+		} = this.props;
+		const guid = media && media.videopress_guid ? media.videopress_guid : null;
+
+		if ( guid ) {
+			updatePoster( guid, { at_time: Math.floor( currentTime ) } );
+		}
+	}
+
+	handleUploadImageClick = () => {
+		isSelectingFrame = false;
+		this.setState( { pauseVideo: true } );
+	}
+
+	handleUploadImage = ( file ) => {
+		if ( ! file ) {
+			return;
+		}
+
+		const {
+			media,
+			updatePoster,
+		} = this.props;
+		const guid = media && media.videopress_guid ? media.videopress_guid : null;
+
+		if ( guid ) {
+			this.props.resetPosterState();
+			updatePoster( guid, { file } );
+		}
+	}
+
+	getVideoEditorProps( poster ) {
+		const { media } = this.props;
+		const videoProperties = { poster };
+
+		if ( media && media.ID ) {
+			videoProperties.ID = media.ID;
+		}
+
+		return videoProperties;
+	}
+
+	renderError() {
+		const {
+			onCancel,
+			translate,
+		} = this.props;
+
+		return (
+			<Notice
+				status="is-error"
+				showDismiss={ true }
+				text={ translate( 'We are unable to edit this video.' ) }
```

Comment:
Hi! I've found a possible matching string that has already been translated **18** times:
`translate( 'We are unable to edit this image.' )` *ES Score: 8*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).